src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/CapacityTest.kt

```diff
@@ -0,0 +1,224 @@
+      .isUnauthorized
+  }
+
+  @Test
+  fun `Get Capacity on non existent Premises returns 404`() {
+    val jwt = jwtAuthHelper.createValidJwt()
+
+    webTestClient.get()
+      .uri("/premises/9054b6a8-65ad-4d55-91ee-26ba65e05488/capacity")
+      .header("Authorization", "Bearer $jwt")
+      .exchange()
+      .expectStatus()
+      .isNotFound
+  }
+
+  @Test
+  fun `Get Capacity with no bookings or lost beds returns OK with empty list body`() {
+    val premises = premisesEntityFactory
+      .withTotalBeds(30)
+      .withYieldedLocalAuthorityArea { localAuthorityEntityFactory.produceAndPersist() }
+      .withYieldedProbationRegion { probationRegionEntityFactory.withYieldedApArea { apAreaEntityFactory.produceAndPersist() }.produceAndPersist() }
+      .produceAndPersist()
+
+    val jwt = jwtAuthHelper.createValidJwt()
+
+    webTestClient.get()
+      .uri("/premises/${premises.id}/capacity")
+      .header("Authorization", "Bearer $jwt")
+      .exchange()
+      .expectStatus()
+      .isOk
+      .expectBodyList<DateCapacity>()
+      .hasSize(0)
+  }
+
+  @Test
+  fun `Get Capacity with booking in past returns OK with empty list body`() {
+    val premises = premisesEntityFactory
+      .withTotalBeds(30)
+      .withYieldedLocalAuthorityArea { localAuthorityEntityFactory.produceAndPersist() }
+      .withYieldedProbationRegion { probationRegionEntityFactory.withYieldedApArea { apAreaEntityFactory.produceAndPersist() }.produceAndPersist() }
+      .produceAndPersist()
+
+    bookingEntityFactory.withDepartureDate(LocalDate.now().minusDays(1))
+      .withYieldedKeyWorker { keyWorkerEntityFactory.produceAndPersist() }
+      .withPremises(premises)
+      .produceAndPersist()
+
+    val jwt = jwtAuthHelper.createValidJwt()
+
+    webTestClient.get()
+      .uri("/premises/${premises.id}/capacity")
+      .header("Authorization", "Bearer $jwt")
+      .exchange()
+      .expectStatus()
+      .isOk
+      .expectBodyList<DateCapacity>()
+      .hasSize(0)
+  }
+
+  @Test
+  fun `Get Capacity with lost beds in past returns OK with empty list body`() {
+    val premises = premisesEntityFactory
+      .withTotalBeds(30)
+      .withYieldedLocalAuthorityArea { localAuthorityEntityFactory.produceAndPersist() }
+      .withYieldedProbationRegion { probationRegionEntityFactory.withYieldedApArea { apAreaEntityFactory.produceAndPersist() }.produceAndPersist() }
+      .produceAndPersist()
+
+    lostBedsEntityFactory.withEndDate(LocalDate.now().minusDays(1))
+      .withPremises(premises)
+      .produceAndPersist()
+
+    val jwt = jwtAuthHelper.createValidJwt()
+
+    webTestClient.get()
+      .uri("/premises/${premises.id}/capacity")
+      .header("Authorization", "Bearer $jwt")
+      .exchange()
+      .expectStatus()
+      .isOk
+      .expectBodyList<DateCapacity>()
+      .hasSize(0)
+  }
+
+  @Test
+  fun `Get Capacity with booking in future returns OK with list entry for each day until the booking ends`() {
```

Comment:
I dunno if it'd be easier and quicker to break all these edge cases into unit tests, which would be cheaper / faster than having integration tests for all the possible cases. What do you think?

Reply:
Yeah agreed, I've removed some of the integration tests and added some unit ones